packages/blade/src/components/Table/TablePagination.web.tsx

```diff
@@ -0,0 +1,422 @@
+import React, { useEffect } from 'react';
+import styled from 'styled-components';
+import getIn from 'lodash/get';
+import { useTableContext } from './TableContext';
+import { tablePagination } from './tokens';
+import BaseBox from '~components/Box/BaseBox';
+import {
+  ChevronLeftIcon,
+  ChevronRightIcon,
+  ChevronsLeftIcon,
+  ChevronsRightIcon,
+  MoreHorizontalIcon,
+} from '~components/Icons';
+import { Dropdown, DropdownOverlay } from '~components/Dropdown';
+import { SelectInput } from '~components/Input/DropdownInputTriggers';
+import { ActionList, ActionListItem } from '~components/ActionList';
+import { Text } from '~components/Typography';
+import { useTheme } from '~utils';
+import { Button } from '~components/Button';
+import { makeAccessible } from '~utils/makeAccessible';
+
+type TablePaginationProps = {
+  /**
+   * The default page size
+   * @default 10
+   **/
+  defaultPageSize?: 10 | 25 | 50;
+  /**
+   * The current page. Passing this prop will make the component controlled and will not update the page on its own.
+   **/
+  currentPage?: number;
+  /**
+   * Callback function that is called when the page is changed
+   */
+  onPageChange?: ({ page }: { page: number }) => void;
+  /**
+   * Callback function that is called when the page size is changed
+   */
+  onPageSizeChange?: ({ pageSize }: { pageSize: number }) => void;
+  /**
+   * Whether to show the page size picker. It will be always be hidden on mobile.
+   * @default true
+   */
+  showPageSizePicker?: boolean;
+  /**
+   * Whether to show the page number selector. It will be always be hidden on mobile.
+   * @default false
+   */
+  showPageNumberSelector?: boolean;
+  /**
+   * Content of the label to be shown in the pagination component
+   * @default `Showing 1 to ${totalItems} Items`
+   */
+  label?: string;
+  /**
+   * Whether to show the label. It will be always be hidden on mobile.
+   * @default false
+   */
+  showLabel?: boolean;
+};
+
+const pageSizeOptions: NonNullable<TablePaginationProps['defaultPageSize']>[] = [10, 25, 50];
+
+const PageSelectionButton = styled.button<{ isSelected?: boolean }>(({ theme, isSelected }) => ({
+  backgroundColor: isSelected
+    ? getIn(theme.colors, tablePagination.pageSelectionButton.backgroundColorSelected)
+    : 'transparent',
+  border: 'none',
+  cursor: 'pointer',
+  height: '32px',
+  width: '32px',
```

Comment:
nit: can be moved to the tokens file as constant + get the 32px from `size` object.

Reply:
Done